Popup/PopupState.js

```diff
@@ -68,6 +69,7 @@ const PopupState = hoc((config, Wrapped) => {	// eslint-disable-line no-unused-v
 		}
 
 		handlePopupHide = () => {
+			forwardClose(null, this.props);
```

Comment:
This is different semantics than we use in moonstone where `onClose` for popup meant that the user requested it close whereas this means that the popup has already closed. Was this intentional?

Reply:
Sort of? This may boil down to not having a clear picture, poor heuristics. Do we need different terminology to make this clearer in the future? `handleCloseing` vs `handleClosed`, etc?

Reply:
naming aside, you want `onClose` to be triggered after the popup closes? I'm not clear on that point.

Reply:
and if so, how would an author decide to prevent closing? (that q just occurred to me now)

Reply:
Ignoring how this code is currently implemented, and not ignoring naming, honestly `onClose` is vague to me. `onClosing` I would assume to be the event to attach to if I wanted something to fire before the Popup has started its close cycle, as in "upon closing this popup, do X`. Whereas `onClosed`, being past tense, would refer to "now that this action has happened, do Y".

I'm not familiar enough with this legacy code to reconcile the above approach with what it's currently doing. Popup was one of the components we never migrated to `ui`, sadly.

Reply:
For the cases I can recall, our convention in Moonstone has been to emit events (with present tense verbs) when the user takes an action (`onClose` before closing, `onToggle` before toggling). We don't have many notification events but there are some (`onHide` and `onShow` from `ui/Transition` are an example).

Note that I'm only noting a deviation from Moonstone and not stating a direction.

Reply:
Present tense for starting an action makes sense to me.